src/Autofac/ILifetimeScope.cs

```diff
@@ -119,15 +119,10 @@ public interface ILifetimeScope : IComponentContext, IDisposable, IAsyncDisposab
         ILifetimeScope BeginLifetimeScope(object tag, Action<ContainerBuilder> configurationAction);
 
         /// <summary>
-        /// Enable tracing (or replace existing tracing) on this scope, routing trace events to the specified tracer.
-        /// All lifetime scopes created from this one will inherit this tracer as well.
+        /// Gets the <see cref="System.Diagnostics.DiagnosticListener"/> to which
+        /// trace events should be written.
         /// </summary>
-        /// <param name="tracer">The <see cref="IResolvePipelineTracer"/> implementation.</param>
-        /// <remarks>
-        /// Only one tracer is supported at once, so calling this will replace any prior tracer on this scope. Nested scopes
-        /// will continue to retain the same trace behaviour.
-        /// </remarks>
-        void AttachTrace(IResolvePipelineTracer tracer);
+        DiagnosticListener DiagnosticSource { get; }
```

Comment:
If someone subscribes to events on the `DiagnosticListener` on ILifetime Scope rather than the Container, they might expect to only capture events for that scope. I know we aren't going to support that pattern, but perhaps we should enforce it by only making the `DiagnosticListener` publicly available on `IContainer` with the `DiagnosticSource` only cropping back up when we start executing resolves?

I can't picture a scenario where someone would need access to write messages to **our** diagnostic source outside of a pipeline.

Reply:
I'll see what I can do here. If I recall correctly, in some cases we only pass the interface around and I need access to the parent source to hand it down to the child. Which is to say, this isn't necessarily only for the consumer; I think I need it to make the hand-off work. But I'll look.

Reply:
Looks like I can add some new constructor parameters to things like `ResolveOperationBase` to allow me to pass in the diagnostic source and this should work.

Reply:
Pushed changes such that only `IContainer` has a diagnostic source property. Everything else is internal. The source gets propagated through constructor parameters as needed.